float64 angle_fb
float64 dist_fb